### PR TITLE
OCPBUGS#5137: Add 0.0.0.0 in /etc/resolv.conf file

### PR DESCRIPTION
**Pull request by xenolinux**
Version(s): 4.12+

Issue: https://issues.redhat.com/browse/OCPBUGS-5137

Link to docs preview: https://57092--docspreview.netlify.app/openshift-enterprise/latest/networking/multiple_networks/configuring-additional-network.html#configuring-additional-network_approaches-managing-additional-network

QE review:
- [x] QE has approved this change.

Additional information:

### CODE REVIEW

**Comment by itzikb-redhat** [comment]
_If_ there is not DNS configured for the subnet of the additional network a dns server with the value of 0.0.0.0  e.g. ```openstack subnet set --dns-nameserver 0.0.0.0 additional_subnet1```

File: networking/multiple_networks/configuring-additional-network.adoc [hunk 1]
```diff
@@ -24,6 +24,11 @@ For an additional network, IP addresses are provisioned through an IP Address Ma
 
 * Applying a YAML manifest: You can manage the additional network directly by creating an `NetworkAttachmentDefinition` object. This approach allows for the chaining of CNI plugins.
 
+[NOTE]
+====
+When you install {product-title} on {rh-openstack-first} by using Open Virtual Network (OVN), a worker node has an additional network with a subnet that has no DNS server address. You must set a DNS server address as `0.0.0.0` in the `/etc/resolv.conf` file on the worker node.
```
Comment (mandre) [inline networking/multiple_networks/configuring-additional-network.adoc:29]: This is not correct. When deploying nodes with multiple interfaces on certain configuration of RHOSP (RHOSP with the OVN SDN), it is possible that the DNS setting coming from the secondary interface takes precedence over the DNS setting coming from the primary one. This becomes problematic if the DNS server can't resolve internal records.
In that case, _the OpenStack admin_ should change the DNS setting for the subnet, or remove DNS settings for the subnet:
```
openstack subnet set --no-dns-nameservers <subnet_id>
```

https://docs.openstack.org/neutron/latest/admin/config-dns-res.html#case-1-each-virtual-network-uses-unique-dns-resolver-s

See also https://issues.redhat.com/browse/OCPBUGS-3998 for a manifestation of this issue.

Comment (mandre) [reply networking/multiple_networks/configuring-additional-network.adoc:29]: @itzikb-redhat please confirm my understanding of the issue is correct.

Comment (xenolinux) [reply networking/multiple_networks/configuring-additional-network.adoc]: @mandre I updated the PR- could you please check?

Comment (xenolinux) [reply networking/multiple_networks/configuring-additional-network.adoc]: @itzikb-redhat Could you PTAL too?

File: networking/multiple_networks/configuring-additional-network.adoc [hunk 2]
```diff
@@ -24,6 +24,15 @@ For an additional network, IP addresses are provisioned through an IP Address Ma
 
 * Applying a YAML manifest: You can manage the additional network directly by creating an `NetworkAttachmentDefinition` object. This approach allows for the chaining of CNI plugins.
 
+[NOTE]
+====
+When deploying {product-title} nodes with the multiple network interfaces on {rh-openstack-first} with OVN SDN, subnet id of the secondary interface might take precedence over the subnet id of the primary interface. In this case, remove the DNS nameserver of the secondary interface's subnet id:
```
Comment (mandre) [inline networking/multiple_networks/configuring-additional-network.adoc:29]:
```suggestion
When deploying {product-title} nodes with multiple network interfaces on {rh-openstack-first} with OVN SDN, DNS settings coming from the secondary interface might take precedence over the ones of the primary interface. In this case, remove the DNS nameservers for the subnet the secondary interface is attached to:
```

File: networking/multiple_networks/configuring-additional-network.adoc [hunk 3]
```diff
@@ -24,6 +24,15 @@ For an additional network, IP addresses are provisioned through an IP Address Ma
...
+[source,terminal]
+----
+$ openstack subnet set --no-dns-nameservers <subnet_id>
```
Comment (mandre) [inline networking/multiple_networks/configuring-additional-network.adoc:32]: I wonder if that's enough, of if you need to explicitly set the DNS nameserver to 0.0.0.0 like @itzikb-redhat was suggesting.
@itzikb-redhat can you confirm?

File: networking/multiple_networks/configuring-additional-network.adoc (hunk 2 above)
Comment (xenolinux) [reply networking/multiple_networks/configuring-additional-network.adoc]: Addressed- changed the wording little bit

File: networking/multiple_networks/configuring-additional-network.adoc (hunk 3 above)
Comment (itzikb-redhat) [reply networking/multiple_networks/configuring-additional-network.adoc:32]: > I wonder if that's enough, of if you need to explicitly set the DNS nameserver to 0.0.0.0 like @itzikb-redhat was suggesting. @itzikb-redhat can you confirm?

If I understand correctly from https://docs.openstack.org/neutron/rocky/admin/config-dns-res.html it should be the same. I checked with setting to 0.0.0.0

Comment (xenolinux) [reply networking/multiple_networks/configuring-additional-network.adoc:32]: @itzikb-redhat @mandre Do we need to change the command? `$ openstack subnet set --no-dns-nameservers <subnet_id>`

Comment (itzikb-redhat) [reply networking/multiple_networks/configuring-additional-network.adoc:32]: I would prefer to set it to 0.0.0.0

File: networking/multiple_networks/configuring-additional-network.adoc [hunk 4]
```diff
@@ -24,6 +24,15 @@ For an additional network, IP addresses are provisioned through an IP Address Ma
 
 * Applying a YAML manifest: You can manage the additional network directly by creating an `NetworkAttachmentDefinition` object. This approach allows for the chaining of CNI plugins.
 
+[NOTE]
+====
+When deploying {product-title} nodes with the multiple network interfaces on {rh-openstack-first} with OVN SDN, DNS configuration of the secondary interface might take precedence over the DNS configuration of the primary interface. In this case, remove the DNS nameservers for the subnet id that is attached to the secondary interface:
+[source,terminal]
+----
+$ openstack subnet set 0.0.0.0 <subnet_id>
```
Comment (itzikb-redhat) [inline networking/multiple_networks/configuring-additional-network.adoc:32]: Please take a look at https://docs.openstack.org/python-openstackclient/pike/cli/command-objects/subnet.html
$ openstack subnet set --dns-nameserver 0.0.0.0 <subnet_id>

Comment (xenolinux) [reply networking/multiple_networks/configuring-additional-network.adoc]: Addressed

File: networking/multiple_networks/configuring-additional-network.adoc [hunk 5]
```diff
@@ -24,6 +24,15 @@ For an additional network, IP addresses are provisioned through an IP Address Ma
 
 * Applying a YAML manifest: You can manage the additional network directly by creating an `NetworkAttachmentDefinition` object. This approach allows for the chaining of CNI plugins.
 
+[NOTE]
+====
+When deploying {product-title} nodes with the multiple network interfaces on {rh-openstack-first} with OVN SDN, DNS configuration of the secondary interface might take precedence over the DNS configuration of the primary interface. In this case, remove the DNS nameservers for the subnet id that is attached to the secondary interface:
```
Comment (JoeAldinger) [inline networking/multiple_networks/configuring-additional-network.adoc:29]:
```suggestion
When deploying {product-title} nodes with multiple network interfaces on {rh-openstack-first} with OVN SDN, DNS configuration of the secondary interface might take precedence over the DNS configuration of the primary interface. In this case, remove the DNS nameservers for the subnet id that is attached to the secondary interface:
```

**Review by JoeAldinger — COMMENTED** [review]
Small suggestion for clarity.

File: networking/multiple_networks/configuring-additional-network.adoc (hunk 5 above)
Comment (xenolinux) [reply networking/multiple_networks/configuring-additional-network.adoc]: Addressed

**Pull request merged**